pallets/rad-claims/src/tests.rs

```diff
@@ -0,0 +1,271 @@
+        let node_3 = RadClaims::sorted_hash_of(&leaf_hash_6, &leaf_hash_7);
+        let node_4 = RadClaims::sorted_hash_of(&leaf_hash_8, &leaf_hash_9);
+        let node_00 = RadClaims::sorted_hash_of(&node_0, &node_1);
+        let node_01 = RadClaims::sorted_hash_of(&node_2, &node_3);
+        let node_000 = RadClaims::sorted_hash_of(&node_00, &node_01);
+        let node_root = RadClaims::sorted_hash_of(&node_000, &node_4);
+
+        let four_sorted_hashes: [H256; 4] = [leaf_hash_3.into(), node_0.into(), node_01.into(), node_4.into()];
+        assert_ok!(RadClaims::store_root_hash(Origin::signed(ADMIN), node_root));
+        assert_eq!(RadClaims::verify_proofs(&USER_B, &amount, &four_sorted_hashes.to_vec()), true);
+    });
+}
+
+#[test]
+fn set_upload_account() {
+    TestExternalitiesBuilder::default().build().execute_with( || {
+        assert_eq!(RadClaims::get_upload_account(), 0x0);
+        assert_err!(RadClaims::set_upload_account(Origin::signed(USER_A), USER_A), BadOrigin);
+        assert_ok!(RadClaims::set_upload_account(Origin::signed(ADMIN), USER_A));
+        assert_eq!(RadClaims::get_upload_account(), USER_A);
+    });
+}
+
+#[test]
+fn store_root_hash() {
+    TestExternalitiesBuilder::default().build().execute_with( || {
+        assert_eq!(RadClaims::get_upload_account(), 0x0);
+        // USER_A not allowed to upload hash
+        let root_hash = <MockRuntime as frame_system::Config>::Hashing::hash(&[0; 32]);
+        assert_err!(
+            RadClaims::store_root_hash(Origin::signed(USER_A), root_hash),
+            Error::<MockRuntime>::MustBeAdmin
+        );
+        // Adding ADMIN as allowed upload account
+        assert_ok!(RadClaims::set_upload_account(Origin::signed(ADMIN), ADMIN));
+        assert_eq!(RadClaims::get_upload_account(), ADMIN);
+        assert_ok!(RadClaims::store_root_hash(Origin::signed(ADMIN), root_hash));
+        assert_eq!(RadClaims::get_root_hash(root_hash), true);
+    });
+}
+
+fn pre_calculate_single_root(
+    account_id: &<MockRuntime as frame_system::Config>::AccountId,
+    amount: &<MockRuntime as pallet_balances::Config>::Balance,
+    other_hash: &<MockRuntime as frame_system::Config>::Hash
+) -> H256 {
+    let mut v: Vec<u8> = account_id.encode();
+    v.extend(amount.encode());
+    let leaf_hash = <MockRuntime as frame_system::Config>::Hashing::hash(&v);
+
+    RadClaims::sorted_hash_of(&leaf_hash, other_hash)
+}
+
+#[test]
+fn claim() {
+    TestExternalitiesBuilder::default().build().execute_with( || {
+        let amount: u128 = 100 * currency::RAD;
+        // Random sorted hashes
+        let one_sorted_hashes: [H256; 1] = [[0; 32].into()];
+
+        // Bad origin, signed vs unsigned
+        assert_noop!(
+            RadClaims::claim(Origin::signed(USER_B), USER_B, amount, one_sorted_hashes.to_vec()),
+            BadOrigin
+        );
+
+        // proof validation error - roothash not stored
+        assert_err!(
```

Comment:
`assert_noop` maybe ?

Reply:
Done... modify `assert_err` to `assert_noop`